aioesphomeserver/device.py

```diff
@@ -79,13 +101,56 @@ def get_entity_by_key(self, key):
             return None
         return self.entities[key - 1]
 
-    async def run(self):
+    async def run(self, api_port, web_port):
         from . import NativeApiServer, WebServer
 
-        self.add_entity(NativeApiServer(name="_server"))
-        self.add_entity(WebServer(name="_web_server"))
+        self.api_port = api_port
+        self.web_port = web_port
+
+        self.add_entity(NativeApiServer(name="_server", port=self.api_port))
+        self.add_entity(WebServer(name="_web_server", port=self.web_port))
+
+        self.zeroconf = await self.register_zeroconf(self.api_port)
 
         async with asyncio.TaskGroup() as tg:
             for entity in self.entities:
                 if hasattr(entity, 'run'):
                     tg.create_task(entity.run())
+
+    async def register_zeroconf(self, port):
```

Comment:
Could you talk a bit about why this is both here and in BasicEntity? It looks largely the same except for a couple differences in the properties.

Reply:
Agree, it's duplicated. I'll move all the zeroconf code to the device file and remove from the basic entity.